test/testers.jl

```diff
@@ -30,6 +32,20 @@ fbtestkws(x, y; err = true) = err ? error() : x
         end
     end
 
+    @testset "test derivative conjugated in pullback" begin
+        ChainRulesCore.frule((_, Δx), ::typeof(sinconj), x) = (sin(x), cos(x) * Δx)
+
+        # define rrule using ChainRulesCore's v0.9.0 convention, conjugating the derivative
+        # in the rrule
+        function ChainRulesCore.rrule(::typeof(sinconj), x)
+            sinconj_pullback(ΔΩ) = (NO_FIELDS, @thunk(conj(cos(x)) * ΔΩ))
```

Comment:
AFAICT thunking this is pointless, unless it is just for testing purposes?
Things that only have a single output that is liable to be used don't benifit from thunking as they will certainly be unthunked.

```suggestion
            sinconj_pullback(ΔΩ) = (NO_FIELDS, conj(cos(x)) * ΔΩ)
```

Reply:
That's right, yes, in case the user did thunk, this makes sure the testing function still works. It's not really an important check, and I could remove the `@thunk`.

Reply:
can just add a comment to that effect